Enable yaml format file

tzmfreedom opened the pull request:
YAML can be used to represent a Swagger specification file.  
This pull request enable to use yaml format swagger file for schema_conformist.  

```ruby
Rails.application.config.schema_conformist.driver = :open_api_2
Rails.application.config.schema_conformist.schema_path = Rails.root.join('doc', 'swagger.yml') # `swagger.yaml` is also available.
```

tzmfreedom committed:
Enable yaml format file

kymmt90 commented on test/driver_test.rb:
```diff
@@ -51,4 +51,29 @@ class DriverTestClass < Object
+
+  test '#schema_hash with json format argument returns correct hash data' do
+    Rails.application.config.schema_conformist.schema_path = Rails.root.join('public', 'swagger.json')
+    json_data = <<-JSON.strip_heredoc
```
How about using `<<~` instead of Active Support's `strip_heredoc`? `<<~` has the same function as `strip_heredoc` and is a Ruby core function after Ruby 2.3.

tzmfreedom replied:
I thought available ruby version for schema_conformist is 2.2 or above.  
Because ruby 2.2 doesn't work `<<~`, I use `strip_heredoc` instead of `<<~`.  
But maintenance of ruby 2.2 will be ended since the end of the March of 2018.  
So I fix from `strip_heredoc` to `<<~`. 

https://github.com/kymmt90/schema_conformist/pull/5/files#diff-b814a1fc6d7f1b214dc94e1bc1169d66

kymmt90 replied:
Rails 5 supports Ruby 2.2.2 or later, so your thoughts make sense 👍  http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#ruby-versions

I think this is only test so it seems that there is no need to be sensitive to Ruby versions 😎

kymmt90 commented on test/driver_test.rb:
```diff
@@ -51,4 +51,29 @@ class DriverTestClass < Object
+
+  test '#schema_hash with yaml format argument returns correct hash data' do
+    Rails.application.config.schema_conformist.schema_path = Rails.root.join('public', 'swagger.yaml')
+    yaml_data = <<-YAML.strip_heredoc
```
Same as above.

kymmt90 reviewed (commented):
@tzmfreedom 

Thank you for your PR! Though CI does not work (it's my fault), I have checked your patch works fine.

Could you add tests to `test/dummy/app/test/controllers` to make sure this gem works in a real app with using YAML? It is OK to reuse the structure of  `test/dummy/app/test/controllers/products_controller_with_open_api_2_test.rb`.

tzmfreedom commented:
> Could you add tests to test/dummy/app/test/controllers to make sure this gem works in a real app with using YAML? It is OK to reuse the structure of test/dummy/app/test/controllers/products_controller_with_open_api_2_test.rb.

OK! I added test code and yaml formatted swagger file.

* https://github.com/kymmt90/schema_conformist/pull/5/files#diff-4842b76d31194de37a8792f039f57b6f
* https://github.com/kymmt90/schema_conformist/pull/5/files#diff-860cb5053a173f4d338133c5d6a0bb60

tzmfreedom commented:
Sometimes, test is not passed.
I think this code causes(not `teardown` but `setup`).
https://github.com/kymmt90/schema_conformist/pull/5/files#diff-b814a1fc6d7f1b214dc94e1bc1169d66R8

May I modify this code from `teardown` to `setup`?
If it's ok, I'll fix it on this pull request.

kymmt90 commented:
> May I modify this code from teardown to setup?
> If it's ok, I'll fix it on this pull request.

OK🙆

tzmfreedom committed:
Fix from teardown to setup because sometimes test is not passed

tzmfreedom commented:
Thank you! I fixed it.

> Could you add tests to test/dummy/app/test/controllers to make sure this gem works in a real app with using YAML? It is OK to reuse the structure of test/dummy/app/test/controllers/products_controller_with_open_api_2_test.rb.

I added test code and yaml formatted swagger file.

* https://github.com/kymmt90/schema_conformist/pull/5/files#diff-4842b76d31194de37a8792f039f57b6f
* https://github.com/kymmt90/schema_conformist/pull/5/files#diff-860cb5053a173f4d338133c5d6a0bb60

kymmt90 reviewed (approved):
LGTM 👌

kymmt90 commented:
Released as v0.2.0! https://github.com/kymmt90/schema_conformist/releases/tag/v0.2.0
